### ISSUE TITLE
添加开关、按键等事件的日志

### ISSUE THREAD

**Issue by mhz20031217** [NJU-ProjectN/nvboard#24]
在开发 [NVDL](https://github.com/mhz20031217/nju-verilated-digital-lab) 时，hwr 想到，如果链接了 nvboard 的二进制文件能够输出按键等信息到日志，那么可以自动生成简单的 testbench （延时问题可以另外写程序解决）。
![hwr 提议](https://github.com/NJU-ProjectN/nvboard/assets/39209322/7e41dc7d-9a45-4f38-a153-de713f8208cb)

**Comment by sashimi-yzh** [comment]
很好的想法, 你们可以先根据开发需要添加日志, 然后提个PR.

**Comment by scratch-er** [comment]
Good idea. But I think it is better to have this feature turned off by default, and it has to be explicitly turned on. Or NVBoard users can be made crazy with the console output messed up and struggle to find a way to turn off log.

There is also an out of box and drop-in method of logging the events: You can specify `$dumpfile$` and `$dumpvars$` in the verilog files to be simulated, and invoke verilog with `--trace` (or specify `TRACE` in `verilate` if you are using CMake). Then the generated executable will produce a VCD waveform. There's no need to change most of the source code. And VCD files can be easily parsed, they are just plain text files. The main drawback of this method is that the VCD files can be huge.
